core/src/services/dashmap/backend.rs

```diff
@@ -70,45 +70,36 @@ impl Builder for DashmapBuilder {
 }
 
 /// Backend is used to serve `Accessor` support in dashmap.
-pub type DashmapBackend = kv::Backend<Adapter>;
+pub type DashmapBackend = typed_kv::Backend<Adapter>;
 
 #[derive(Debug, Clone)]
 pub struct Adapter {
-    inner: DashMap<String, Vec<u8>>,
+    inner: DashMap<String, typed_kv::Value>,
 }
 
 #[async_trait]
-impl kv::Adapter for Adapter {
-    fn metadata(&self) -> kv::Metadata {
-        kv::Metadata::new(
-            Scheme::Dashmap,
-            &format!("{:?}", &self.inner as *const _),
-            Capability {
-                read: true,
-                write: true,
-                scan: true,
-                ..Default::default()
-            },
-        )
+impl typed_kv::Adapter for Adapter {
+    fn metadata(&self) -> (Scheme, String) {
```

Comment:
We need to return a `typed_kv::Metadata` here in which carries a `typed_kv::Capability {get: true, set: true, scan: true}`.

Reply:
Perhaps the capability check will be reintroduced in a later PR.

Reply:
I note that https://github.com/apache/incubator-opendal/pull/2215 seems to be relevant.

Reply:
> I note that #2215 seems to be relevant.

Not related. Please ignore it.

Reply:
> Perhaps the capability check will be reintroduced in a later PR.

No, they are different things. `typed_kv` must provide it's own Capability so that `typed_kv::Backend` can implement `Accessor::info()` correctly.

Reply:
Got it and I'll add it in some time later.